saito-core/src/core/routing_controller.rs

```diff
@@ -0,0 +1,403 @@
+use std::sync::Arc;
+use std::time::Duration;
+
+use async_trait::async_trait;
+use log::{debug, info, trace};
+use tokio::sync::mpsc::Sender;
+use tokio::sync::RwLock;
+
+use crate::common::command::{GlobalEvent, NetworkEvent};
+use crate::common::defs::SaitoHash;
+use crate::common::interface_io::InterfaceIO;
+use crate::common::keep_time::KeepTime;
+use crate::common::process_event::ProcessEvent;
+use crate::core::data;
+use crate::core::data::block::{Block, BlockType};
+use crate::core::data::blockchain::Blockchain;
+use crate::core::data::configuration::Configuration;
+use crate::core::data::msg::block_request::BlockchainRequest;
+use crate::core::data::msg::message::Message;
+use crate::core::data::peer::Peer;
+use crate::core::data::peer_collection::PeerCollection;
+use crate::core::data::storage::Storage;
+use crate::core::data::wallet::Wallet;
+use crate::core::blockchain_controller::MempoolEvent;
+use crate::core::miner_controller::MinerEvent;
+
+#[derive(Debug)]
+pub enum BlockchainEvent {
```

Comment:
Rename to RoutingEvent ?

Reply:
Yeah, missed that one, new updates pushed